cmd/bucket-handlers.go

```diff
@@ -507,11 +524,13 @@ func (api objectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 	// Write success response.
 	writeSuccessResponseXML(w, encodedSuccessResponse)
 	for _, dobj := range deletedObjects {
-		if dobj.DeleteMarkerReplicationStatus == string(replication.Pending) || dobj.VersionPurgeStatus == Pending {
-			globalReplicationState.queueReplicaDeleteTask(DeletedObjectVersionInfo{
-				DeletedObject: dobj,
-				Bucket:        bucket,
-			})
+		if replicateDeletes {
+			if dobj.DeleteMarkerReplicationStatus == string(replication.Pending) || dobj.VersionPurgeStatus == Pending {
```

Comment:
this status would be pending only if object qualifies for delete marker replication and would not even get queued for buckets where delete marker replication is not enabled.

Reply:
The problem is a small bug here can cause dire consequences, you shouldn't use a mutating data structure to block a channel.

The problem is pending is added elsewhere, because right after this PR the blocking is observed in deletes.

Reply:
pending is set before the Delete operation, and passed back to the resultant DeletedObjects array from the erasure layer. The only time this will block is if the replicationDelete channel queue is full (10k), and that is a design issue that will affect only buckets with delete marker replication enabled. I tested for a non replication setup with master , and this never hung/blocked - primarily because it wouldn't be queued for replication. imo, this PR is cosmetic and not fixing whatever hang you observed.

Reply:
Hmm interesting then there is an issue somewhere else lurking which needs to be addressed @poornas